src/function/table/arrow.cpp

```diff
@@ -294,8 +294,7 @@ void ArrowTableFunction::ArrowScanFunctionParallel(ClientContext &context, const
 }
 
 idx_t ArrowTableFunction::ArrowScanMaxThreads(ClientContext &context, const FunctionData *bind_data_p) {
-	auto &data = (const ArrowScanFunctionData &)*bind_data_p;
-	return data.stream->number_of_batches;
+	return context.db->NumberOfThreads();
```

Comment:
The stream still has a number of rows, right? Can we use that as a proxy here if it is set, e.g. one thread every 100K rows?